doc/user/content/transform-data/patterns/rules-engine.md

```diff
@@ -0,0 +1,149 @@
+---
+title: "Rules execution engine"
+description: "Encode rules as data and apply them using lateral joins."
+aliases:
+  - /guides/rules-engine/
+  - /sql/patterns/rules-engine/
+menu:
+  main:
+    parent: 'sql-patterns'
+---
+
+SQL queries are often used to represent "rules" and evaluate which data in a dataset adhere to those rules.
+In Materialize, the results of these queries can be kept automatically up-to-date, enabling use cases like real-time, rules-based alerting.
+However, some organizations have a lot of rules.
+This can lead users to ask questions like "Can Materialize support 100,000 materialized views"?
+
+Fortunately there is a pattern to address this need without creating zillions of separate views.
+If your rules are simple enough to be expressed as data (i.e. not arbitrary SQL), then you can use lateral joins to implement a rules execution engine.
```

Comment:
```suggestion
With Materialize, you can express your rules as data (instead of arbitrary SQL) and use lateral joins to create a rules execution engine.
```